src/herder/TxQueueLimiter.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "herder/TxQueueLimiter.h"
+#include "crypto/ShortHash.h"
+#include "util/GlobalChecks.h"
+
+namespace stellar
+{
+
+namespace
+{
+uint64
+getIDFromAccountID(AccountID const& accountID)
+{
+    return stellar::shortHash::computeHash(stellar::ByteSlice(
```

Comment:
Maybe it's overkill I agree, I can remove it as it's only for a few thousand transactions.
Reason here (not clear because I forgot to move some code, read on) is we potentially have a "walk all transactions" degenerate case in `canAddTx` (all transactions have a different fee), and we're looking up the `id` in all fee groups (note: the *computation* of the id should really be done outside the loop).

I am realizing that I forgot to use the `UnorderedSet/Map` templates instead of the `std` versions, so I need to touch that code anyways

Reply:
removed and moved to proper types